react-components/table-intro/step8.md

```diff
@@ -1,118 +1,64 @@
 
-3) Add a function "updateList" just above handlePerPageSelect to handle setting the rows based on when the numPerPage state property changes.
+Add the following code to the App component, just below the previous state property you created. Supply a `1` as the default value so that the table intuitively loads with the first page worth of data.
 
-<pre class="file" data-target="clipboard">
-const updateList = value => {
-  setCurrentPage(value);
-  const beginMark = (value - 1) * numPerPage;
-  const endMark = beginMark + numPerPage;
-  let newRows = defaultRows.slice(beginMark, endMark);
-  setRows(newRows);
-};
+<pre class="file">
+const [currentPage, setCurrentPage] = React.useState(1);
 </pre>
 
-4) Inside the handlePerPageSelect handler, make a call to updateList and pass it the currentPage state property.
+4) <strong>Set the `page` prop of the `<Pagination />` component to the `currentPage` state property.</strong>
 
 <pre class="file">
-updateList(currentPage);
+page={currentPage}
 </pre>
 
-5) Add a useEffect hook inside your component that we'll teach to only run when certain properties change. This effects hook will ultimately ensure `numPerPage` and `rows` stay in sync. Inside this effect, make a call to handlePerPageSelect, pass null for the first param (event) and pass numPerPage as the second param. Place this code just above the return statement in your App component.
+5) <strong>Set the Pagination component dropdown options.</strong>
+
+Set the `perPageOptions` prop of the `<Pagination />` component so that we have two options in the dropdown, 2 and 3. This controls how many rows per page are shown.
 
 <pre class="file">
-React.useEffect(() => {
-  handlePerPageSelect(null, numPerPage);
-}, []);
+perPageOptions={[{ title: "2", value: 2 }, { title: "3", value: 3 }]}
 </pre>
 
-6) Locate the empty dependency array for the useEffect hook you just created, it's the second parameter (`[]`) that was passed to useEffect. This second parameter to useEffect represents an array of the function's dependencies.
+Notice the "per page options" dropdown now reflect options of 2 and 3 per page.
 
-7) Specify numPerPage and handlePerPageSelect as <a href="https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect" target="_blank">effect dependencies</a> by listing them in the dependency array from the previous step. Replace the empty array with the following;
+<img src="table-intro/assets/step-8-perPageOptions-complete.png" alt="Image of what table looks like after step 8, sub-step 5." style="box-shadow: rgba(3, 3, 3, 0.2) 0px 1.25px 2.5px 0px;" />
+
+6) <strong>Create a function to handle updating the number of rows per page.</strong>
+
+Add the following code inside the App component definition (just below the state properties you added earlier). Inside this function update the `numPerPage` state property to the value selected in the dropdown by calling `setNumPerPage` and passing it the `newPerPage` parameter.
```

Comment:
remove you and when you say "added earlier" reference the step where it was added.